src/ClientHandle.cpp

```diff
@@ -1033,7 +1033,7 @@ void cClientHandle::HandleLeftClick(int a_BlockX, int a_BlockY, int a_BlockZ, eB
 
 		if (
 			cChunkDef::IsValidHeight(BlockY) &&
-			cBlockHandler::For(m_Player->GetWorld()->GetBlock({ BlockX, BlockY, BlockZ })).IsClickedThrough()
+			(m_Player->GetWorld()->GetBlock({ BlockX, BlockY, BlockZ }) == E_BLOCK_FIRE)
```

Comment:
The client does the bounding box calculations and sends us the block its clicking on all of the time... only apart from fire, where it appears to send us the block behind the fire.

However I agree this check shouldn't be here, BlockInfo is better because they've added other fire types recently.